src/HTTPClientFactory.ts

```diff
@@ -29,8 +29,8 @@ export class HTTPClientFactory {
      * @returns A HTTPClient instance
      */
     public static createWithAuthenticationProvider(authProvider: AuthenticationProvider): HTTPClient {
-        let authenticationHandler = new AuthenticationHandler(authProvider);
-        let httpMessageHandler = new HTTPMessageHandler();
+        const authenticationHandler = new AuthenticationHandler(authProvider);
+        const httpMessageHandler = new HTTPMessageHandler();
```

Comment:
Can we chat about possibly adding the HttpMessageHandler in the createWithMiddleware method?  In .Net the final handler is treated specially.  Conceptually it isn't treated as middleware but the end of the pipeline.